src/authgss_hash.c

```diff
@@ -70,8 +70,9 @@ static struct authgss_hash_st authgss_hash_st = {
 static inline uint64_t
 gss_ctx_hash(gss_union_ctx_id_desc *gss_ctx)
 {
-	return ((uint64_t) gss_ctx->mech_type +
-		(uint64_t) gss_ctx->internal_ctx_id);
+	/* mash together two pointers, could be more sophisticated? */
+	return ((uint64_t)((uintptr_t)gss_ctx->mech_type +
+			   (uintptr_t)gss_ctx->internal_ctx_id));
```

Comment:
It will still overflow for two 64-bit numbers with the top bit set. Should do add with end-around-carry, as in assembly. But I'll make your simpler change, and expand the comment to explain. All we care about here is the number is unique, or has a very low collision rate.